lib/ASTGen/Sources/ASTGen/CompilerBuildConfiguration.swift

```diff
@@ -321,6 +334,133 @@ public func freeConfiguredRegions(
   UnsafeMutableBufferPointer(start: regions, count: numRegions).deallocate()
 }
 
+/// Cache used when checking for inactive code that might contain a reference
+/// to specific names.
+private struct InactiveCodeContainsReferenceCache {
+  let syntax: SourceFileSyntax
+  let configuredRegions: ConfiguredRegions
+}
+
+/// Search in inactive/unparsed code to look for evidence that something that
+/// looks unused is actually used in some configurations.
+private enum InactiveCodeChecker {
+  case name(String)
```

Comment:
Should this take an `Identifier` already instead of a `String`?

```suggestion
  case name(Identifier)
```

Reply:
`Identifier` needs either `SyntaxText` in an arena, or a `StaticString`. I don't really have either here because I'm coming from a bridged `StringRef`.